### PR TITLE
truncate volume name when it exceed 63 chars

### PR DESCRIPTION
**Pull request by khevse**
Failed to create a pvc(gluster)
```
Warning

ProvisioningFailed  3s    persistentvolume-controller  
Failed to provision volume with StorageClass "gluster": 
failed to create volume: failed to create endpoint/service monitoring/glusterfs-dynamic-prometheus-mon-prometheus-db-prometheus-mon-prometheus-0: 
error creating service: Service "glusterfs-dynamic-prometheus-mon-prometheus-db-prometheus-mon-prometheus-0" is invalid: 
metadata.name: Invalid value: "glusterfs-dynamic-prometheus-mon-prometheus-db-prometheus-mon-prometheus-0": 
must be no more than 63 characters
```

### CODE REVIEW

File: pkg/k8sutil/k8sutil.go [hunk 1]
```diff
@@ -203,3 +204,64 @@ func SanitizeVolumeName(name string) string {
 	}
 	return strings.Trim(name, "-")
 }
+
+// TruncateVolumeName truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec)
+func TruncateVolumeName(name string) string {
```
Comment (squat) [inline pkg/k8sutil/k8sutil.go]: This function duplicates most of the functionality from the `SanitizeVolumeName` func above. That func makes a volume name DNS1123 compliant. These funcs should either be reconciled or, we should just use that one wholesale where this new one was used.

Comment (khevse) [reply pkg/k8sutil/k8sutil.go]: I agree. I fixed my mistake.

**Comment by leeyan85** [comment]
Is the issue resolved ? It makes me mad

**Comment by khevse** [comment]
> Is the issue resolved ? It makes me mad
Good. I will test another solution and report the result.

**Comment by khevse** [comment]
A conflict can appear with other names after truncating. Upgrade is not planned.

**Pull request closed without merging**